### PR TITLE
Review Client Side

### PR DESCRIPTION
**Pull request by JeniaSavenko**

### CODE REVIEW

File: src/api/socket.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+} from '../actions/PostActions';
+import { Auth } from '../constants/Auth';
+import { SocketConst } from '../constants/Socket';
+
+let socket;
+let store;
+
+class Socket {
+  static socket;
+
```
Comment (MeyerPV) [inline src/api/socket.js:11]: redundantly spacing

File: src/api/socket.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import io from 'socket.io-client';
+import { AsyncStorage } from 'react-native';
+import {
+  createPostTitle, finishEdit, getPosts, rmPost, savePost, startEdit,
+} from '../actions/PostActions';
+import { Auth } from '../constants/Auth';
+import { SocketConst } from '../constants/Socket';
+
+let socket;
```
Comment (MeyerPV) [inline src/api/socket.js]: remove

File: src/screen/PostScreen.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  };
+
+  const renderItem = ({ item }) => (
+    <ListItem
+      editMode={item.editMode}
+      userId={item.editing}
+      title={item.title}
+      text={item.text}
+      updatedAt={item.updatedAt}
+      onPress={() => {
```
Comment (MeyerPV) [inline src/screen/PostScreen.js]: move to separate function

File: src/components/LogOutButton.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import { logoutUser } from '../actions/UserActions';
+
+const LogOutButton = ({ logout }) => {
+  const { t: translate } = useTranslation();
+  return (
+    <Block width row ac js>
+      <Block mh={10}>
+        <Button
+          title={translate('logout')}
+          onPress={() => logout(NavigationActions.navigate)}
```
Comment (MeyerPV) [inline src/components/LogOutButton.js:16]: separate function

File: src/screen/CreateUserScreen.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  navigation,
+  createUser,
+}) => {
+  const [userName, setUserName] = useState();
+
+  const [userPass, setUserPass] = useState();
+
+  const { t } = useTranslation();
+
+  function handleUserNameChange(name) {
```
Comment (MeyerPV) [inline src/screen/CreateUserScreen.js:18]: it's new trend write all functions in old format?

File: index.js [hunk 6]
```diff
@@ -2,8 +2,8 @@
  * @format
  */
 
-import {AppRegistry} from 'react-native';
+import { AppRegistry } from 'react-native';
 import App from './App';
-import {name as appName} from './app.json';
+import { name as appName } from './app.json';
```
Comment (esnts) [inline index.js:7]: 💩

File: src/api/socket.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    if (Socket.instance) {
+      return Socket.instance;
+    }
+    Socket.instance = this;
+
+    return this;
+  }
+
+
+  configureSocket = (s) => {
```
Comment (esnts) [inline src/api/socket.js:29]: 💩 what is the 's'? I think that 's' is socks, is it?

File: src/api/socket.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    store = s;
+  };
+
+  runSocket = () => {
+    socket = io(this.url, {
+      query: {
+        token: AsyncStorage.getItem(Auth.userToken),
```
Comment (esnts) [inline src/api/socket.js:36]: can you get it from redux store and use redux-persis for saving?

File: src/api/socket.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+  editModeFinish = (postId) => {
+    store.dispatch(finishEdit(postId));
+    socket.emit(SocketConst.editModeFinish, postId);
+  };
+}
+
+const WebSocket = new Socket(Auth.mainUrl);
+
+WebSocket.init();
```
Comment (esnts) [inline src/api/socket.js]: 💩

File: src/components/ButtonWithIcon.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+      outline
+      icon={(
+        <Icon
+          name={iconName}
+          type={Style.iconType}
+          color={Style.colorWhite}
+        />
+      )}
+      titleStyle={Style.titleStyle}
+      title={t(title)}
```
Comment (esnts) [inline src/components/ButtonWithIcon.js]: Are u really thinking that it is a good idea to use translation here ?

File: src/components/HeaderButton.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { Button } from 'react-native-elements';
+import { useTranslation } from 'react-i18next';
+import Block from './Block';
+
```
Comment (esnts) [inline src/components/HeaderButton.js:5]: spacing

File: src/components/ListItem.js [hunk 12]
```diff
@@ -1,32 +1,47 @@
...
-          {title}
-        </Title>
-        <Body>
-          {text}
-        </Body>
+const ListItem = ({
+  title, text, updatedAt, onPress, editMode, userId,
+}) => {
+  const { t } = useTranslation();
+  const newDate = new Date(updatedAt).toISOString().slice(0, 10);
```
Comment (esnts) [inline src/components/ListItem.js]: ну и что блять это вообще такое?

**Pull request closed without merging**